feat: implement user avatar upload and replace functionality

mohammed4122002 opened the pull request:
<img width="2196" height="760" alt="لقطة شاشة 2025-12-28 115147" src="https://github.com/user-attachments/assets/0ed86c36-103d-435f-a8cf-857e8234cdec" />
<img width="2166" height="1085" alt="لقطة شاشة 2025-12-28 115154" src="https://github.com/user-attachments/assets/f2b1684e-2036-4bcb-a99b-f6cf3ec457d4" />
<img width="2194" height="826" alt="لقطة شاشة 2025-12-28 115318" src="https://github.com/user-attachments/assets/7b513b40-0b74-493f-8936-5fa90d56d23f" />
<img width="2242" height="1003" alt="لقطة شاشة 2025-12-28 115325" src="https://github.com/user-attachments/assets/72ba0947-c596-4a2f-8432-9e7bbf15741d" />
<img width="948" height="153" alt="لقطة شاشة 2025-12-28 115510" src="https://github.com/user-attachments/assets/51d4d14b-7725-4c64-bb7e-9a18b484c8b2" />
<img width="1490" height="750" alt="لقطة شاشة 2025-12-28 120647" src="https://github.com/user-attachments/assets/3bbace7b-bf83-4595-8e02-bbbd89d59578" />

mohammed4122002 committed:
feat: implement user avatar upload and replace functionality

mohammed4122002 committed:
feat: implement user avatar upload and replace functionality

waidfj commented on src/app/uploads/infrastructure/controllers/uploads.controller.ts:
```diff
@@ -0,0 +1,86 @@
+
+
+
+
```
نزل اكستنشن prettier عشان تلتزم بقواعد ترتيب الكود

waidfj commented on src/app.module.ts:
```diff
@@ -38,6 +39,7 @@ const ENV = process.env.NODE_ENV;
     }),
 
     UserModule,
+     UploadsModule,
```
```suggestion
     FileModule,
```

waidfj commented on src/app.module.ts:
```diff
@@ -7,6 +7,7 @@ import { UserModule } from '@app/user/infrastructure/modules/user.module';
 import jwtConfig from '@config/jwt.config';
 import { APP_GUARD } from '@nestjs/core';
 import { TokenGuard } from '@app/user/infrastructure/guards/token.guard';
+import { UploadsModule } from '@app/uploads/infrastructure/modules/uploads.module';
```
```suggestion
import { UploadsModule } from '@app/shared/uploads/infrastructure/modules/uploads.module';
```
اعمل folder سميه shared عشان نحط فيه الmodules اللي راح يستخدمها كل الapplication، زي هادا الmodule

waidfj commented on src/app/uploads/application/dtos/upload-avatar.dto.ts:
```diff
@@ -0,0 +1,6 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UploadAvatarDto {
```
```suggestion
export class UploadImageRequestDto {
```
عشان نخليها مناسبة لاي صورة بشكل عام

waidfj commented on src/app/uploads/application/dtos/upload-avatar.dto.ts:
```diff
@@ -0,0 +1,6 @@
+
+export class UploadAvatarDto {
+  @ApiProperty({ type: 'string', format: 'binary', description: 'Avatar image file' })
+  avatar: any;
```
؟؟

waidfj commented on src/app/uploads/infrastructure/controllers/uploads.controller.ts:
```diff
@@ -0,0 +1,86 @@
+
+@ApiTags('Users')
+@Controller('user/profile')
+export class UploadsController {
```
هادا الكلام مسؤولية الuser controller

waidfj commented on src/app/uploads/infrastructure/modules/uploads.module.ts:
```diff
@@ -0,0 +1,12 @@
+    imports: [
+      
+        UserModule,
+    ],
```
لا الUserModule بيعمل import لل FileModule وبيستخدمه عشان يكمل الservice تبعته
الFileModule هيكون موديول بيعطي سيرفيس الفايل لكل الابليكيشن، مش هينفع نخليه يعمل امبورت لكل المشروع ويشتغل على كل اشي لانه هادا هيزيد من تعقيد المشروع

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,45 @@
+
+    try {
+    
+      const baseUrl = process.env.APP_URL || 'http://localhost:3002'; 
```
استخدم الconfig

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,45 @@
+    try {
+    
+      const baseUrl = process.env.APP_URL || 'http://localhost:3002'; 
+      const avatarUrl = `${baseUrl}/uploads/avatars/${file.filename}`;
```
```suggestion
      const avatarUrl = `${baseUrl}/files/images/user_avatars/${file.filename}`;
```
اعمل فايل للconsts اللي زي هيك

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,45 @@
+export class UpdateUserAvatarUsecase {
+  constructor(@Inject('UsersRepository') private readonly userRepository: IUserRepository) {}
+
+  async execute(user: User, file: FileData): Promise<{ avatar: string }> {
```
ليش ما استخدمت الrequestDto ؟

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,45 @@
+export class UpdateUserAvatarUsecase {
+  constructor(@Inject('UsersRepository') private readonly userRepository: IUserRepository) {}
+
+  async execute(user: User, file: FileData): Promise<{ avatar: string }> {
```
```suggestion
  async execute(user: User, file: FileData): Promise<void> {
```

waidfj replied:
الupdate ترجع no content ايش رايك؟

mohammed4122002 replied:
تمام هذا  أنظف

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,45 @@
+      user.avatar = avatarUrl;
+      user.updatedAt = new Date();
+
+      const updated = await this.userRepository.save(user);
```
ليش عملت يوزر جديد؟

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,45 @@
+        filename: file.filename 
+      });
+
+      return { avatar: updated.avatar! };
```
```suggestion
```

waidfj reviewed (commented):
ضيف الfolder اللي راح تتخزن فيه الfiles لgitignore

AbdellatifEllouze commented on src/app/shared/uploads/application/dtos/upload-file.dto.ts:
أظن ملوش لازمة عشان بقيدنا كثير مثلاً لو اردنا نعمل upload file مصاحب لمعلومات أخرى مو رح نستطع إستخدمو

AbdellatifEllouze commented on src/app/user/application/dtos/requests/update-avatar.dto.ts:
هذا يكفي أظن ممكن نستغنى عن   UploadFileRequestDto إحذف إل user كمان من هنا إل dto هي نسخة من input أو output json

AbdellatifEllouze commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,24 @@
+    private readonly configService: ConfigService,
+  ) { }
+
+  async execute({ file, user }: UpdateAvatarDto): Promise<{ avatar: string }> {
```
```suggestion
  async execute(file: File, user : User): Promise<{ avatar: string }> {
```

waidfj replied:
ليش راح نستغني عن الdto؟

AbdellatifEllouze commented on src/app/shared/uploads/application/usecases/upload-file-usecase.ts:
هذا ماهو usecase ممكن ايكون service ، إل usecase هو senario يقوم به user في حالتنا هذي هو تغير الصورة ممكن لل-usecase انادي ال-service أخر ليقوم بمهمة معينة كما فعلت

AbdellatifEllouze commented on src/app/shared/uploads/application/usecases/upload-file-usecase.ts:
```diff
@@ -0,0 +1,19 @@
+import { UploadFileRequestDto } from '../dtos/upload-file.dto';
+
+@Injectable()
+export class UploadFileUsecase {
```
الفكرة إنو ممكن نعمل service أخر في المستقبل يحفض الملفات في S3 مثلاً 
```suggestion
export class UploadFileService implement IUplaodFile {
```

AbdellatifEllouze commented on src/app/shared/uploads/application/usecases/upload-file-usecase.ts:
```diff
@@ -0,0 +1,19 @@
+
+@Injectable()
+export class UploadFileUsecase {
+  async execute({ subFolder, file }: UploadFileRequestDto): Promise<string> {
```
بما أنك قمت بإنشاء module جديد و يكون فيه UploadFileService أظن إنو إل module في حد ذاتو هو إلي احدد مكان التنزيل ممكن نعمل fileTypeEnum فيها حالياً avatar ممكن أندف أشيء أخرى في إل مستقبل ونعمل mapper وعلى حسب إل type نحدد المكان

AbdellatifEllouze replied:
```suggestion
  async execute(file: File, type: FileTypeEnum): Promise<string> {
```

AbdellatifEllouze commented on src/app/shared/uploads/application/usecases/upload-file-usecase.ts:
```diff
@@ -0,0 +1,19 @@
+    const outPath = join(destFolder, filename);
+
+    await pipeline(file.file, createWriteStream(outPath));
+    return filename;
```
```suggestion

    return filename;
```

AbdellatifEllouze commented on src/app/shared/uploads/application/dtos/upload-file.dto.ts:
ممكن تعمل module جديد يهتم file uploader

waidfj commented on src/app/shared/uploads/application/dtos/upload-file.dto.ts:
```diff
@@ -0,0 +1,15 @@
+  @ApiProperty({
+    type: 'string',
+    format: 'binary',
+    description: 'The file to upload'
```
```suggestion
    description: 'The file object.',
```

waidfj commented on src/app/shared/uploads/application/dtos/upload-file.dto.ts:
```diff
@@ -0,0 +1,15 @@
+// src/app/uploads/application/dtos/upload-image.dto.ts
```
```suggestion
```

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,24 @@
+
+    const filename = await this.uploadFileUsecase.execute({ subFolder: "avatars", file: file });
+    const baseUrl = this.configService.get<string>('app.url');
+    const avatarUrl = `${baseUrl}/uploads/avatars/${filename}`;
```
افصلها لان ممكن اكتر من يوزكيس تستخدم هادا الurl
`    const avatarUrl = `${this.configService.get<string>('app.url')}/${filename}`;`
او 
`    const avatarUrl = this.filePathService.create(filename);`

mohammed4122002 committed:
chore: saloony-15 resolve merge conflicts and update configurations

MohaAAlJ committed:
Feat/task#57 add user status column (#65)

Sajaalghoul committed:
Feature/city country schema (#61)

rozanashaqqoura committed:
feat(token): create token module and entity (#55)

mohammed4122002 committed:
Feature/start notification module (#58)

MohaAAlJ committed:
Feat#59(auth): implement hybrid authorization with salon roles (#66)

mohammed4122002 committed:
feat: implement notifier core and channel routing (#68) (#69)

Sajaalghoul committed:
chore(format): run Prettier on address files and migration

Run Prettier to normalize formatting for the address entity, schema, and the new migration file.

Sajaalghoul committed:
feat(address): PROJ-33 add address entity migration and schema

Sajaalghoul committed:
feat(address): PROJ-33 add address entity migration and schema

Sajaalghoul committed:
feat(address): PROJ-33 add address entity migration and schema

Sajaalghoul committed:
ci: use --legacy-peer-deps in CI to avoid ERESOLVE during npm ci

Sajaalghoul committed:
Merge pull request #75 from salooony/address-entity-33

Address entity 33

Sajaalghoul committed:
created the country creation api (#77)

* feat(country): add country repository port

* feat(country): add create country request and response DTOs

* feat(country): add country mapper

* feat(address): create country repository

* feat(country): add create country use case

* feat(address): add create country controller (admin only)

* feat(address): wire AddressModule for country creation

MohaAAlJ committed:
feat/Task#70: Implement Request Email Verification Code Endpoint (#78)

MohaAAlJ committed:
feat: Implement request phone verification endpoint

MohaAAlJ committed:
feat: Implement request phone verification endpoint

MohaAAlJ committed:
refactor: update phone verification to use 204 status and ConflictException

rozanashaqqoura committed:
feature/saloony-63-token-generator-engine (#79)

* feat: implement token generator engine

* feat: implement token generator engine

---------

Co-authored-by: Abdellatif Ellouze <abdellatif.ellouze@gmail.com>

MohaAAlJ committed:
Resolved merge conflicts

MohaAAlJ committed:
refactor:use dynamic TokenService & fetch SMS template from DB

MohaAAlJ committed:
refactor(user): address Abdellatif & Mohammed reviews - dynamic template, logging, BadRequestException

MohamedEllouze committed:
Merge pull request #81 from salooony/feature/SALOONY-071-request-phone-verification

feat/Task#71: Implement request phone verification endpoint

mohammed4122002 committed:
feat(address): SAL-26 update country API (PUT)

mohammed4122002 committed:
feat(address): SAL-27 List Countries API (GET)

waidfj committed:
Apply suggestion from @waidfj

waidfj committed:
Update src/app/address/application/usecases/update-country.usecase.ts

Commit:
fix quality

waidfj committed:
Merge pull request #87 from salooony/Update-Country-API-26

feat(address): SAL-26 update country API (PUT)

mohammed4122002 committed:
chore: resolve merge conflicts

mohammed4122002 committed:
feat(address): add country filters and normalize codes

mohammed4122002 committed:
fix(address): handle optional country activity filter

waidfj committed:
Merge pull request #88 from salooony/List-Countries-API-27

feat(address): SAL-27 List Countries API (GET)

MohaAAlJ committed:
refactor(user): update enums and defaults

MohaAAlJ committed:
fix(test): remove deleted usecase from user spec

MohaAAlJ committed:
refactor(user): update enums and defaults

MohaAAlJ committed:
refactor(user): update enums and defaults

waidfj committed:
Merge pull request #93 from salooony/clean-code-incomplete-task

refactor/Task#92: update enums and defaults

waidfj commented on src/app/user/application/dtos/requests/update-avatar.dto.ts:
```diff
@@ -0,0 +1,10 @@
+    format: 'binary',
+    description: 'The file object',
+  })
+  file: any;
```
ليش any؟

waidfj commented on src/app/shared/uploads/application/services/upload-file.service.ts:
```diff
@@ -0,0 +1,25 @@
+import { FileType } from '../../domain/enums/file-type.enum';
+
+@Injectable()
+export class UploadFileService implements IUploadFileService {
```
```suggestion
export class UploadFileService implements IUploadFile {
```

mohammed4122002 replied:
تمام

waidfj commented on src/app/shared/uploads/application/services/file-path.service.ts:
```diff
@@ -0,0 +1,13 @@
+export class FilePathService {
+  constructor(private readonly configService: ConfigService) {}
+
+  getFileUrl(type: FileType, filename: string): string {
```
```suggestion
  getFileUrl(filename: string, type: FileType): string {
```
يمكن هيك الترتيب منطقي اكتر، ايش رايك؟

waidfj replied:
وايش رايك نسميها createFileUrl احسن؟ عشان نمنع اللخبطة انا تلخبطت فيها قبل شوية

mohammed4122002 replied:
اوكي

waidfj replied:
نسيت تغير الترتيب
+ايش رايك تخليها generate 😂

waidfj commented on src/app/user/infrastructure/controllers/user.controller.ts:
```diff
@@ -76,43 +68,18 @@ export class UserController {
 
-  @ApiOperation({ summary: 'Delete current user account.' })
+  @Put('profile/avatar')
+  @ApiOperation({ summary: 'Update user avatar' })
   @ApiBearerAuth()
-  @ApiResponse({
-    status: HttpStatus.NO_CONTENT,
-    description: 'Account deleted successfully.',
-  })
-  @ApiResponse({
-    status: HttpStatus.UNAUTHORIZED,
-    description: 'User should be logged in.',
-  })
-  @ApiResponse({
-    status: HttpStatus.INTERNAL_SERVER_ERROR,
-    description: 'Something went wrong, try again.',
-  })
-  @Delete('/me')
-  @HttpCode(HttpStatus.NO_CONTENT)
-  async deleteMe(@CurrentUser() user: User): Promise<void> {
-    await this.deleteUserUseCase.execute(user.id);
-  }
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({ type: UpdateAvatarDto })
```
كمل الswagger documentation

mohammed4122002 replied:
بس هل تستغني عن UpdateAvatarDto لان عبد اللطيف  حكا : 
أظن ملوش لازمة عشان بقيدنا كثير مثلاً لو اردنا نعمل upload file مصاحب لمعلومات أخرى مو رح نستطع إستخدمو

waidfj replied:
ممكن dto لكل service ؟

waidfj commented on src/app/shared/uploads/domain/enums/file-type.enum.ts:
```diff
@@ -0,0 +1,3 @@
+export enum FileType {
+  AVATAR = 'avatars',
```
```suggestion
  IMAGE = 'image',
```
على اساس بعدين نعمل فيديو وكذا؟ ولا نستخدم الامتداد احسن؟ (زي .png, .jpg, ...)

mohammed4122002 replied:
نعمل مثل هيك 
export enum FileType {
  AVATAR = 'avatars',
  IMAGE = 'images',
  VIDEO = 'videos',
} 
توافقي الراي ؟

waidfj replied:
طيب ما هو الavatar اصلا image

mohammed4122002 replied:
نوع من انواع التمييز ! بس ما في مشكلة بنخليها Image

waidfj commented on src/app/shared/uploads/application/services/file-path.service.ts:
```diff
@@ -0,0 +1,13 @@
+
+  getFileUrl(type: FileType, filename: string): string {
+    const baseUrl = this.configService.get<string>('app.url');
+    return `${baseUrl}/uploads/${type}/${filename}`;
```
```suggestion
    return `${this.configService.get<string>('app.url')}/uploads/${type}/${filename}`;
```

mohammed4122002 replied:
تمت

waidfj commented on src/app/shared/uploads/application/services/upload-file.service.ts:
```diff
@@ -0,0 +1,25 @@
+
+@Injectable()
+export class UploadFileService implements IUploadFileService {
+  async execute(type: FileType, file: MultipartFile): Promise<string> {
```
بيلزم documentation للفنكشن هاي بالcomments

mohammed4122002 replied:
اوكي تمام

waidfj replied:
كان قصدي للexecution تعت الفنكشن نفسها

waidfj commented on src/app/shared/uploads/infrastructure/modules/file.module.ts:
```diff
@@ -0,0 +1,15 @@
+@Module({
+  providers: [
+    {
+      provide: 'IUploadFileService',
```
اعمللها const زي اللي صرنا نعملهم

waidfj commented on src/app/shared/uploads/infrastructure/modules/file.module.ts:
```diff
@@ -0,0 +1,15 @@
+    },
+    FilePathService,
+  ],
+  exports: ['IUploadFileService', FilePathService],
```
ليش الmodules التاني هتستخدم الfilePathService ؟

mohammed4122002 replied:
حاليًا مستخدمة من UserModule عبر UpdateUserAvatarUsecase، لذلك لازم تكون موجودة في providers وexports داخل FileModule

mohammed4122002 replied:
فايدة FilePathService هي centralizing logic تبع بناء روابط الملفات. بدل ما UpdateUserAvatarUsecase يركب الـ avatar URL يدويا نخلي هذا المنطق في خدمة واحدة قابلة لإعادة الاستخدام لو أضفنا ملفات أخرى لاحقا

waidfj commented on src/app/user/application/dtos/requests/update-avatar.dto.ts:
```diff
@@ -0,0 +1,10 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UpdateAvatarDto {
+  @ApiProperty({
+    type: 'string',
+    format: 'binary',
+    description: 'The file object',
+  })
```
ضيف كمان swagger documentation

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -0,0 +1,27 @@
+import { IUploadFileService } from '@app/shared/uploads/domain/ports/iupload-file.service';
+import { FilePathService } from '@app/shared/uploads/application/services/file-path.service';
+import { FileType } from '@app/shared/uploads/domain/enums/file-type.enum';
+import { MultipartFile } from '@app/shared/uploads/domain/interfaces/multipart-file.interface';
```
```suggestion
import { IUserRepository } from '@user/domain/ports/iuser.repository';
import { User } from '@user/domain/entities/user';
import { IUploadFileService } from '@shared/uploads/domain/ports/iupload-file.service';
import { FilePathService } from '@shared/uploads/application/services/file-path.service';
import { FileType } from '@shared/uploads/domain/enums/file-type.enum';
import { MultipartFile } from '@shared/uploads/domain/interfaces/multipart-file.interface';
```

waidfj commented on src/app/user/infrastructure/controllers/user.controller.ts:
```diff
@@ -1,32 +1,24 @@
+import { BadRequestException, Inject } from '@nestjs/common';
+import { UpdateUserAvatarUsecase } from '@app/user/application/usecases/update-user-avatar.usecase';
+import { UpdateAvatarDto } from '../../application/dtos/requests/update-avatar.dto';
+import { Public } from '@app/user/application/decorators/public.decorator';
```
```suggestion
import { UpdateUserAvatarUsecase } from '@user/application/usecases/update-user-avatar.usecase';
import { UpdateAvatarDto } from '../../application/dtos/requests/update-avatar.dto';
import { Public } from '@user/application/decorators/public.decorator';
```

waidfj commented on src/app/user/infrastructure/mappers/user.mapper.ts:
```diff
@@ -18,14 +18,18 @@ export class UserMapper {
 
+    // preserve id for updates so TypeORM performs an UPDATE instead of INSERT
+    if (user.id) {
+      (entity as any).id = user.id;
```
طيب ليش as any ؟

waidfj commented on src/app/user/infrastructure/mappers/user.mapper.ts:
```diff
@@ -36,7 +40,13 @@ export class UserMapper {
-    entity.deletedAt = user.deletedAt;
+
+    if (user.createdAt) {
+      entity.createdAt = user.createdAt;
+    }
+    if (user.updatedAt) {
```
```suggestion
    }
    
    if (user.updatedAt) {
```

waidfj commented on src/app/user/infrastructure/modules/user.module.ts:
```diff
@@ -13,16 +13,32 @@ import jwtConfig from '@config/jwt.config';
-import { DeleteUserAccountUseCase } from '@app/user/application/usecases/delete-user-account.usecase';
-import { MailerModule } from '@nestjs-modules/mailer';
+import { UpdateUserAvatarUsecase } from '@app/user/application/usecases/update-user-avatar.usecase';
+import { FileModule } from '@app/shared/uploads/infrastructure/modules/file.module';
```
```suggestion
import { UpdateUserAvatarUsecase } from '@user/application/usecases/update-user-avatar.usecase';
import { FileModule } from '@shared/uploads/infrastructure/modules/file.module';
```

waidfj commented on src/app/user/infrastructure/modules/user.module.ts:
```diff
@@ -13,16 +13,32 @@ import jwtConfig from '@config/jwt.config';
+      defaults: {
+        from: '"No Reply" <noreply@example.com>',
+      },
+    }),
```
ايش دخله بالfile ؟

mohammed4122002 replied:
هيه موجوده بالفعل بسطر 38 بس عدلت لانها بتعطي ايورو

waidfj commented on src/app/user/infrastructure/schemas/user.entity.ts:
```diff
@@ -1,4 +1,4 @@
-import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn, DeleteDateColumn } from 'typeorm';
+import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';
 
 @Entity({ name: 'users' })
```
ليش؟

waidfj commented on src/types/fastify-multipart.d.ts:
```diff
@@ -0,0 +1,5 @@
+declare module '@fastify/multipart' {
+  import { FastifyPluginAsync } from 'fastify';
+  const fastifyMultipart: FastifyPluginAsync<any>;
+  export default fastifyMultipart;
```
```suggestion
  import { FastifyPluginAsync } from 'fastify';
  
  const fastifyMultipart: FastifyPluginAsync<any>;
  
  export default fastifyMultipart;
```

waidfj commented on uploads/avatars/1768725950466-962972562.png:
اعمل gitignore

mohammed4122002 committed:
feat: implement user avatar upload and update functionality

mohammed4122002 committed:
feat: implement user avatar upload and update functionality

mohammed4122002 committed:
feat: Complete the instructions

mohammed4122002 committed:
feat(user): add avatar upload support

Updates upload handling, Swagger docs, and related configuration.

mohammed4122002 committed:
feat(user): add avatar upload support

Updates upload handling, Swagger docs, and related configuration.

waidfj commented on src/app/user/application/usecases/login.usecase.ts:
```diff
@@ -8,14 +8,14 @@ import {
 import { LoginResponseDto } from '../dtos/responses/login.response.dto';
 import { HashingProviderInterface } from '../providers/hashing.provider.interface';
-import { IUserRepository } from '@app/user/domain/ports/iuser.repository';
+import { IUserRepository, USERS_REPOSITORY } from '@app/user/domain/ports/iuser.repository';
```
```suggestion
import { IUserRepository, USERS_REPOSITORY } from '@user/domain/ports/iuser.repository';
```

waidfj commented on src/app/user/application/usecases/get-user-info.usecase.ts:
```diff
@@ -1,10 +1,10 @@
 import { UserResponseDto } from '../dtos/responses/user.response.dto';
 import { Inject, NotFoundException } from '@nestjs/common';
-import { IUserRepository } from '@app/user/domain/ports/iuser.repository';
+import { IUserRepository, USERS_REPOSITORY } from '@app/user/domain/ports/iuser.repository';
```
```suggestion
import { IUserRepository, USERS_REPOSITORY } from '@user/domain/ports/iuser.repository';
```

waidfj commented on src/app/user/application/usecases/create.usecase.ts:
```diff
@@ -1,6 +1,6 @@
 import { ConflictException, Inject, InternalServerErrorException } from '@nestjs/common';
 import { UserTransformer } from '../transformers/user.transformer';
-import { IUserRepository } from '@app/user/domain/ports/iuser.repository';
+import { IUserRepository, USERS_REPOSITORY } from '@app/user/domain/ports/iuser.repository';
```
```suggestion
import { IUserRepository, USERS_REPOSITORY } from '@user/domain/ports/iuser.repository';
```

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -1,27 +1,26 @@
-    const filename = await this.uploadFileService.execute(FileType.AVATAR, file);
-    const avatarUrl = this.filePathService.getFileUrl(FileType.AVATAR, filename);
+  async execute(file: MultipartFile, user: User): Promise<void> {
+    const filename = await this.uploadFile.execute(FileType.AVATAR, file);
```
اتفقنا انه الavatar عبارة عن image ؟

waidfj commented on src/app/user/infrastructure/guards/token.guard.ts:
```diff
@@ -1,6 +1,6 @@
 import { CanActivate, ExecutionContext, Inject, Injectable, UnauthorizedException } from '@nestjs/common';
 import { JwtService } from '@nestjs/jwt';
-import { IUserRepository } from '@app/user/domain/ports/iuser.repository';
+import { IUserRepository, USERS_REPOSITORY } from '@app/user/domain/ports/iuser.repository';
```
```suggestion
import { IUserRepository, USERS_REPOSITORY } from '@user/domain/ports/iuser.repository';
```

mohammed4122002 replied:
شكراا عالدقة

waidfj commented on src/test/user.spec.ts:
```diff
@@ -6,6 +6,7 @@ import { CreateUserUsecase } from '@app/user/application/usecases/create.usecase
 import { ForgotPasswordUseCase } from '@app/user/application/usecases/forgot-password.usecase';
 import { GetUserInfoUsecase } from '@app/user/application/usecases/get-user-info.usecase';
 import { LoginUsecase } from '@app/user/application/usecases/login.usecase';
+import { USERS_REPOSITORY } from '@app/user/domain/ports/iuser.repository';
```
```suggestion
import { USERS_REPOSITORY } from '@user/domain/ports/iuser.repository';
```

waidfj commented on src/app/user/application/usecases/update-user-avatar.usecase.ts:
```diff
@@ -1,27 +1,26 @@
+import { MultipartFile } from '@shared/uploads/domain/interfaces/multipart-file.interface';
 
 @Injectable()
 export class UpdateUserAvatarUsecase {
```
```suggestion
export class UpdateAvatarUsecase {
```
ممكن الاسم يقتصر على هيك بما ان الmodule مختص باليوزر

mohammed4122002 committed:
feat: implement user avatar upload and replace functionalit

mohammed4122002 committed:
feat(user): enhance user avatar upload functionality and clean up code

mohammed4122002 committed:
chore: update package-lock.json

mohammed4122002 committed:
fix: add eslint-disable for fastify unsafe operations

mohammed4122002 committed:
style: format code with prettier
